doc/CMakeLists.txt

```diff
@@ -17,6 +17,23 @@
 # You should have received a copy of the GNU General Public License
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
+if ( NOT CMAKE_CROSSCOMPILING )
+  add_custom_target( generate_help ALL )
+  add_custom_command( TARGET generate_help POST_BUILD
+    COMMAND python -B generate_help.py "${PROJECT_SOURCE_DIR}" "${PROJECT_BINARY_DIR}"
+    WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/extras/help_generator"
+    )
+  install( DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/help"
+    DESTINATION "${CMAKE_INSTALL_DOCDIR}"
+    )
+  install( CODE
+    "execute_process(
+      COMMAND python -B generate_helpindex.py \"${CMAKE_INSTALL_DOCDIR}\"
+      WORKING_DIRECTORY \"${PROJECT_SOURCE_DIR}/extras/help_generator\"
+    )"
```

Comment:
Actually it is not the same code. The code in `add_custom_command()` extracts the help during the `make` step and puts the result in `doc/help` in the current build directory together with a `helpindex.html` containing all help files in that directors. This operation is performed by the script `generate_help.py`. The code in `execute_process()` re-generates the help index so that it contains *all* help files from the global install directory and not just the ones generated in the last call. The script used here is `generate_helpindex.py`. This two-step process is needed to allow modules to install their help to the global help directory and still have an up-to-date index. Otherwise the index would just contain the help files generated by the last call.

I've added comments to the corresponding `CMakeLists.txt` files to clarify this in d739c2a.